posydon/binary_evol/DT/step_detached.py

```diff
@@ -1926,6 +1926,19 @@ def get_star_profile(star, htrack, m0):
                     else:
                         binary.state = "RLO1"
                         binary.event = "oRLO1"
+                
+                if (binary.star_1.state in LIST_ACCEPTABLE_STATES_FOR_HeStar 
+                    and binary.star_2.state in STAR_STATES_H_RICH 
+                    and binary.state == "RLO1"):
```

Comment:
A performance improvement: I'd first check for the "RLO1" before checking the star states.